### PR TITLE
Connected preloader and added ability to refresh current time without api request and ability to refresh token if it is expired 

### PR DESCRIPTION
**Pull request by DmitriyKrivovyaz**

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -7,20 +7,14 @@ import CityForecastView from './views/cityForecastView/CityForecastView';
...
-  return [theme, setTheme];
-};
+import Preloader from './components/Preloader/Preloader';
+import { showPreloaderTimeout } from './globalConsts';
+import { useLocalStorageTheme } from './hooks/hooks';
 
 function App(props) {
   const [theme, setTheme] = useLocalStorageTheme(themes.light);
   const bgImage = bgImages[theme];
+  const [needToShowPreloader, setNeedToShowPreloader] = useState(true);
```
Comment (JaneEllison) [inline client/src/App.jsx]: maybe this isn't the best name for this value

Comment (anton-vlasik) [reply client/src/App.jsx]: 1) shouldShowPreloader
2) isPreloaderShown
3) shown/show -> displayed/display --> opened/open etc

File: client/src/api/weatherApi.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { urls } from '../globalConsts';
 
 export const getAndSetAccessToken = async () => {
   try {
+    configAxiosInterceptors();
```
Comment (anton-vlasik) [inline client/src/api/weatherApi.js]: configure ?

File: client/src/services/locationService.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const getCurrentLocation = async () => {
-  return new Promise(resolve =>
-    navigator.geolocation.getCurrentPosition(position => resolve(position))
-  );
+  return new Promise(resolve => {
+    const position = navigator.geolocation.getCurrentPosition(position => resolve(position));
```
Comment (anton-vlasik) [inline client/src/services/locationService.js]: how do you use position const?

Comment (DmitriyKrivovyaz) [reply client/src/services/locationService.js]: I don't use it. Will replace

**Review by anton-vlasik — APPROVED** [review]

**Review by ximet — APPROVED** [review]

**Pull request merged**